### PR TITLE
Refactoring and readme update

### PR DESCRIPTION
**Pull request by kushti**
In this PR:

* readme update
* refactoring (unused imports and code removals, code reordering etc)

### CODE REVIEW

**Review by greenhat — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -114,14 +114,11 @@ which will run Project Import Wizard. The wizard will use SBT configuration
 You can open `File / Project Structure...` dialog to see project configuration.
 If everything is successful you can compile the project in IDE. 
 
-## FAQ
-[Frequently Asked Questions](FAQ.md)
+## Contributions
 
-## Acknowledgements
+Ergo is open-source and open movement, always in need for testers developers! Please feel free
```
Comment (pragmaxim) [inline README.md:119]: testers `and` developers

**Review by pragmaxim — APPROVED** [review]

**Review by aslesarenko — COMMENTED** [review]
LGTM, please see comments

File: README.md (hunk 1 above)
Comment (kushti) [reply README.md]: fixed

File: src/main/scala/org/ergoplatform/modifiers/history/extension/ExtensionCandidate.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.ergoplatform.settings.Algos
+import scorex.crypto.authds.LeafData
+import scorex.crypto.authds.merkle.{Leaf, MerkleProof, MerkleTree}
+import scorex.crypto.hash.Digest32
+import scorex.util.ModifierId
+
+/**
+  * Extension block section with not filled header id
+  */
+class ExtensionCandidate(val fields: Seq[(Array[Byte], Array[Byte])]) {
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/modifiers/history/extension/ExtensionCandidate.scala:19]: done

**Pull request merged**